pkg/root/signing_config.go

```diff
@@ -38,7 +38,7 @@ type Service struct {
 	URL                 string
 	MajorAPIVersion     uint32
 	ValidityPeriodStart time.Time
-	ValidityPeriodEnd   time.Time
+	ValidityPeriodEnd   *time.Time
```

Comment:
It's recommend to pass `time.Time` as a struct, and check `time.IsZero` to see if it's initialized - https://pkg.go.dev/time#Time

We follow this same pattern in trusted root, checking if it's [zero](https://github.com/sigstore/sigstore-go/blob/db889a109b0deb1494e03d6da27c8348a2d815da/pkg/tlog/entry.go#L282) on use.

Reply:
`isZero()` does not work: that's the bug I'm fixing:
 * The protobuf `Timestamp{0, 0}` means "1970-01-01", golang `time.Time{0, 0}` means "01-01-01"
 * We can keep using the magic value approach, but must then use "1970-01-01" as the magic value which seems a bit icky. I can drop the last commit to achieve that.

Reply:
Oh I see. Do we have the same bug in TrustedRoot then?

Reply:
I think the end timestamp typically does not matter in  verification use case: even if we are past the `ValidityPeriodEnd` of a specific tlog, we still want to use the log to verify entries from that log.

I'm not sure there is any situation where a normal client is actually interested in `ValidityPeriodEnd` from trustedroot.